pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -167,7 +169,13 @@ func getMachineHealthCheckListOptions() *client.ListOptions {
 
 func remediate(r *ReconcileMachineHealthCheck, machine *mapiv1.Machine) (reconcile.Result, error) {
 	glog.Infof("Initialising remediation logic for machine %s", machine.Name)
-	if isMaster(*machine, r.client) {
+
+	infra, err := getClusterInfrastructure(r.client)
```

Comment:
can this be fetch only once at instantiation time?

Reply:
done